node/task.ts

```diff
@@ -725,36 +728,105 @@ export function resolve(...pathSegments: any[]): string {
 export const which = im._which;
 
 /**
- * Returns array of files in the given path, or in current directory if no path provided.  See shelljs.ls
- * @param  {string}   options  Available options: -R (recursive), -A (all files, include files beginning with ., except for . and ..)
+ * Returns array of files in the given path, or in current directory if no path provided.
  * @param  {string[]} paths    Paths to search.
+ * @param  {boolean} recursive Whether to recurse into subdirectories
+ * @param  {boolean} allFiles  Whether to copy all files, include files beginning with ., except for . and ..
  * @return {string[]}          An array of files in the given path(s).
  */
-export function ls(options: string, paths: string[]): string[] {
-    if (options) {
-        return shell.ls(options, paths);
-    } else {
-        return shell.ls(paths);
-    }
+export function ls(paths: string[], recursive: boolean = false, allFiles: boolean = false): string[] {
+    let filePaths: string[] = [];
+    paths.forEach(path => {
```

Comment:
Check for null on paths

Reply:
I think we actually are ok leaving this out. If path is null it will call `dir  /b [/s] [/a]` or `ls [-R] [-A]` which will return directory info for the current directory. That seems like the behavior we'd want (its equivalent to passing in a space or period)